lib/providers/storage.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:concordium_wallet/entities/accepted_terms_and_conditions.dart';
+import 'package:concordium_wallet/state/network.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+
+/// Service for interacting with [Hive].
+class StorageProvider {
+  final Box<AcceptedTermsAndConditions> _acceptedTermsAndConditionBox;
+
+  const StorageProvider._(this._acceptedTermsAndConditionBox);
+
+  static Future<StorageProvider> init() async {
+    await Hive.initFlutter();
+    _registerAdapters();
+    await _openBoxes();
+
+    return StorageProvider._(await Hive.openBox<AcceptedTermsAndConditions>(AcceptedTermsAndConditions.table));
+  }
+
+  /// Register all adapters needed for typed boxes.
+  static void _registerAdapters() {
+    Hive.registerAdapter(AcceptedTermsAndConditionsAdapter());
+  }
+
+  /// Opens all boxes asynchronously.
+  static Future<void> _openBoxes() async {
+    final atcFuture = Hive.openBox<AcceptedTermsAndConditions>(AcceptedTermsAndConditions.table);
```

Comment:
I made this class to show how boxes should be open asynchronously when more are added.
I will keep for now since we then avoid to refactor the structure.